### ISSUE TITLE
helm3 list --all-namespaces by default using .helmrc

### ISSUE THREAD

**Issue by rafis** [helm/helm#7535]
User Story: As Kubernetes cluster SysOps I want to specify 
```
list:
  all-namespaces-by-default: true
```
in `~/.helmrc` and now when I execute
```
helm3 list
```
then I want it to be acted like if I have executed:
```
helm3 list --all-namespaces
```
so that I can ease out migration process from Helm v2 to v3 for our users.

Output of `helm version`:
3.0.3

Output of `kubectl version`:
1.12+

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
kubeadm installed cluster

**Comment by marckhouzam** [comment]
Be aware that it's not just `helm list` which now requires a namespace but all other cluster commands such as `helm delete`, `helm get`, `helm upgrade` etc.

So for your usecase, it may turn out to be more confusing for your users if you made `helm list` pretend to behave like it used to, but all the other commands have the new behaviour. And to be complete, it is not possible to make the other commands behave like helm 2 because with helm 3 you can have releases with the same name in different namespaces.

But I do see where your coming from. What I've done myself is present the helm 3 changes to my team, likened them to `kubectl` existing behaviour, and offered the idea of using an alias for `helm ls --all-namespaces`.

**Comment by mattfarina** [comment]
This is me musing for a moment...

Handling global configuration would be nice. I don't think we would want a `.helmrc` file off the home directory because that doesn't work with our XDG setup very well. It might be useful to have a method to set/get/delete configuration from the `helm` CLI and store that configuration in the home directory alongside the repositories configuration.

The `-A` flag can be used for short. Notice the capitalization. This is what I prefer so I can type less.

Making some of these settings globally configurable is going to be nuanced. For example, I could see someone using the all namespaces flag to list and then accidentally installing/updating an app to the wrong namespace. I would want to avoid that as Marc noted.

I wonder how something could be done that provides a safe and good UX for this kind of thing.

**Comment by marckhouzam** [comment]
Good point about the shorter `-A` flag. It'll be coming out soon in helm 3.1.

**Comment by hickeyma** [comment]
@marckhouzam What do you mean about the `-A` flag? Was this added in #6848?

**Comment by marckhouzam** [comment]
@hickeyma yep. `helm list -A`

**Comment by bridgetkromhout** [comment]
Hi, @rafis - As you saw above, `helm list -A` has been available since 3.1.0. Hopefully this solves for your use case. I'm closing this issue because we now have a recommended path forward for proposals: if you'd like to formally propose further enhancements, please file a [Helm Improvement Proposal](https://github.com/helm/community/blob/master/hips/hip-0001.md) to discuss it with the community. Thanks!
